mlflow/gateway/config.py

```diff
@@ -418,20 +421,46 @@ def validate_limit(cls, value):
 
         return value
 
-    def to_route(self) -> "Route":
-        return Route(
+    def _to_legacy_route(self) -> "_LegacyRoute":
+        return _LegacyRoute(
             name=self.name,
             route_type=self.endpoint_type,
-            model=RouteModelInfo(
+            model=EndpointModelInfo(
                 name=self.model.name,
                 provider=self.model.provider,
             ),
             route_url=f"{MLFLOW_GATEWAY_ROUTE_BASE}{self.name}{MLFLOW_QUERY_SUFFIX}",
             limit=self.limit,
         )
 
+    def to_endpoint(self) -> "Endpoint":
+        from mlflow.deployments.server.config import Endpoint
+
+        return Endpoint(
+            name=self.name,
+            endpoint_type=self.endpoint_type,
+            model=EndpointModelInfo(
+                name=self.model.name,
+                provider=self.model.provider,
+            ),
+            endpoint_url=f"{MLFLOW_GATEWAY_ROUTE_BASE}{self.name}{MLFLOW_QUERY_SUFFIX}",
+            limit=self.limit,
+        )
 
-class RouteModelInfo(ResponseModel):
+
+class RouteDestinationConfig(ConfigModel):
+    name: str
+    traffic_percentage: int
+
+
+class RouteConfig(ConfigModel):
+    name: str
+    task_type: EndpointType
+    destinations: list[RouteDestinationConfig]
+    routing_strategy: Literal["TRAFFIC_SPLIT"] = "TRAFFIC_SPLIT"
```

Comment:
q: Is it just to be future-proof? Do we plan to implement any other strategy soon?

Reply:
in future, we might have other strategy to support (like round-robin)